twist_teleoperation:
  joystick_frame_id:
    type: string
    default_value: ""
    description: "Frame ID for the spacemouse"
  base_frame_id:
    type: string
    default_value: "base_link"
    description: "Frame ID for the robot arm base"
  end_effector_frame_id:
    type: string
    default_value: "gripper_link"
    description: "Frame ID for the robot arm end effector"

  joystick:
    home_button:
      type: int
      default_value: 3
      description: "Button to home the end effector"

  home:
    joint_controller_name:
      type: string
      default_value: "joint_trajectory_controller"
      description: "Name of the joint controller to use for homing"
    joint_names:
      type: string_array
      default_value: []
      description: "Names of the joints to home"
    joint_positions:
      type: double_array
      default_value: []
      description: "Positions of the joints to home"

  update_rate:
    type: int
    default_value: 100
    description: "Rate at which to publish target pose in Hz"

  cartesian_controller_name:
    type: string
    default_value: "cartesian_motion_controller"
    description: "Name of the Cartesian controller to use for teleoperation"

  bounds:
    enabled:
      type: bool
      default_value: false
      description: "Enable boundary limits for the end effector position based on the base frame"
    x_min:
      type: double
      default_value: -0.5
      description: "Minimum X position for the end effector in meters"
    x_max:
      type: double
      default_value: 0.5
      description: "Maximum X position for the end effector in meters"
    y_min:
      type: double
      default_value: -0.5
      description: "Minimum Y position for the end effector in meters"
    y_max:
      type: double
      default_value: 0.5
      description: "Maximum Y position for the end effector in meters"
    z_min:
      type: double
      default_value: 0.0
      description: "Minimum Z position for the end effector in meters"
    z_max:
      type: double
      default_value: 1.0
      description: "Maximum Z position for the end effector in meters"
